### PR TITLE
MS: Segment endpoints

### PR DESCRIPTION
**Pull request by mgz**
https://github.com/attribution/attribution/pull/1014 should be merged.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -170,12 +170,6 @@ Attribution.prototype.send = function(path, msg, fn) {
 
   var projectId = this.options.project || window.Attribution.projectId;
 
-  // If we're POSTing, let's send the project_id in the params
-  // only on POST though, or else we'll get two question marks
-  if(cors)
-    url += "?project_id=" + projectId;
-
-  // If we're JSONPing, let's send it in the encoded params
   msg.project_id = projectId;
```
Comment (troex) [inline lib/index.js]: @mgz let's use `projectId` when sending to our endpoint, backend still should read both but I'd like to keep it here `camelCase` so it's unified with the rest of segment spec

**Comment by mgz** [comment]
@troex done, I copied segmentio/index.js and adapted it.
Now we have tests: `yarn test`.
Now we `retryQueue`: https://github.com/attribution/analytics.js-integration-attribution/blob/88e8d8863c6dbe45e1b4f6b7f3f4b9b457a2b63f/lib/index.js#L60

File: lib/index.js [hunk 2]
```diff
@@ -113,7 +227,10 @@ Attribution.prototype.onidentify = function(identify) {
  */
 
 Attribution.prototype.ontrack = function(track) {
-  this.send('/track', track.json());
+  var json = track.json();
+  // TODO: figure out why we need traits.
+  delete json.traits;
```
Comment (troex) [inline lib/index.js:232]: @mgz we don't need it, it's left over from early implementation, safe to remove this delete

File: lib/index.js [hunk 3]
```diff
@@ -140,18 +258,42 @@ Attribution.prototype.onalias = function(alias) {
...
+Attribution.prototype.normalize = function(message) {
+  var msg = message;
   this.debug('normalize %o', msg);
   var user = this.analytics.user();
   var global = exports.global;
   var query = global.location.search;
-  var ctx = msg.context = msg.context || msg.options || {};
+  var ctx = (msg.context = msg.context || msg.options || {});
   delete msg.options;
+  msg.projectId = this.options.project || window.Attribution.projectId;
```
Comment (troex) [inline lib/index.js:269]: @mgz I don't think we need this `window.Attribution.projectId`, that's probably an old hack which we don't need anymore.

**Pull request merged**